packages/@aws-cdk/aws-events/lib/api-destination.ts

```diff
@@ -0,0 +1,129 @@
+import * as crypto from 'crypto';
+import { Duration, Lazy, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IConnection } from './connection';
+import { CfnApiDestination } from './events.generated';
+
+/**
+ * Supported HTTP operations.
+ */
+export enum HttpMethod {
+
+  /** POST */
+  POST = 'POST',
+  /** GET */
+  GET = 'GET',
+  /** HEAD */
+  HEAD = 'HEAD',
+  /** OPTIONS */
+  OPTIONS = 'OPTIONS',
+  /** PUT */
+  PUT = 'PUT',
+  /** PATCH */
+  PATCH = 'PATCH',
+  /** DELETE */
+  DELETE = 'DELETE',
+
+}
+
+/**
+ * The event Api Destination properties
+ */
+export interface ApiDestinationProps {
+  /**
+   * The ARN of the connection to use for the API destination
+   */
+  readonly connection: IConnection;
+  /**
+   * A description for the API destination.
+   *
+   * @default - none
+   */
+  readonly description?: string;
+  /**
+   * The method to use for the request to the HTTP invocation endpoint.
+   */
+  readonly httpMethod: HttpMethod;
+  /**
+   * The URL to the HTTP invocation endpoint for the API destination..
+   */
+  readonly invocationEndpoint: string;
+  /**
+   * The maximum number of requests per second to send to the HTTP invocation endpoint.
+   */
+  readonly invocationRateLimit: Duration;
+  /**
+   * The name for the API destination.
+   * @default - A unique name will be generated from the construct ID
+   */
+  readonly apiDestinationName: string;
+}
+
+/**
+ * Define an EventBridge Api Destination
+ *
+ * @resource AWS::Events::ApiDestination
+ */
+export class ApiDestination extends Resource {
+  /**
+   * The Connection to associate with Api Destination
+   */
+  public readonly connection: IConnection;
+
+  /**
+   * The Name of the Api Destination created.
+   * @attribute
+   */
+  public readonly apiDestinationName: string;
+
+  /**
+   * The ARN of the Api Destination created.
+   * @attribute
+   */
+  public readonly apiDestinationArn: string;
+
+  constructor(scope: Construct, id: string, props: ApiDestinationProps) {
+    super(scope, id, {
+      physicalName: props.apiDestinationName || Lazy.string({
```

Comment:
Would this work as well ? 

```ts
physicalName: props.apiDestinationName || Lazy.string({ produce: () => Names.uniqueId(this) })
```

The `generateUniqueName()` method seems to limit the name to 21 characters max but I cannot see requirements for this in the CFN doc.